### PR TITLE
Add --analyze option for the linters

### LINKED ISSUE #8792: [Feature Request] Add --analyze option to `pod lib lint`
Add a `--analyze` option to `pod lib lint` to include static analysis when linting. 

Some ideas for implementation at https://stackoverflow.com/a/39635929

### PR DESCRIPTION
**Pull request by paulb777**
Close #8792

### CODE REVIEW

**Comment by segiddins** [comment]
This seems like a new feature, it probably shouldn't be targeting the stable branch?

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -662,7 +666,19 @@ def build_pod
           if scheme.nil?
             UI.warn "Skipping compilation with `xcodebuild` because target contains no sources.\n".yellow
           else
-            output = xcodebuild('build', scheme, 'Release')
+            if analyze
+              output = xcodebuild('analyze', scheme, 'Release')
+              find_output = Executable.execute_command('find', [validation_dir, '-name', '*.html'], false)
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: Is this a reliable way to determine whether analysis failed? Is there a reason why the `xcodebuild` call won't itself fail?

Comment (paulb777) [reply lib/cocoapods/validator.rb]: Unfortunately not, the `xcodebuild` succeeds if the build succeeds even if there are `analyze` issues. See the discussion at https://stackoverflow.com/a/39635929.

**Comment by paulb777** [comment]
Yep, it's a new feature. It seems like a safe, isolated addition that could provide value sooner in 1.7.0 with low risk.  I'm not sure of the exact policy and wanted to discuss if it could go in. I'll retarget to master if it's too late for 1.7.0.

File: lib/cocoapods/validator.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/cocoapods/validator.rb]: should we move this into an `analyze_pod` method for consistency with `build_pod` and `test_pod`?

**Comment by dnkoutso** [comment]
Given we are converging on 1.7.0 I recommend we push that to master. Seems less risky I know but no reason to push it imo.

File: lib/cocoapods/validator.rb (hunk 1 above)
Comment (paulb777) [reply lib/cocoapods/validator.rb]: I'm not sure, since analyze is mostly an option to build_pod.

**Comment by paulb777** [comment]
Rebased to master

File: lib/cocoapods/validator.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/cocoapods/validator.rb]: Gotcha makes sense

**Comment by paulb777** [comment]
I didn't see anything close to what would be needed in the tests for other lint options, so I likely would need some pointers.

**Comment by dnkoutso** [comment]
See `it 'runs xcodebuild with correct arguments when skipping import validation'`

it prints out the command line arguments expected.

**Pull request merged**